client/signup/config/flows-pure.js

```diff
@@ -18,6 +18,14 @@ export function generateFlows( {
 	getDIFMSiteContentCollectionDestination = noop,
 } = {} ) {
 	const flows = [
+		{
+			name: HOSTING_LP_FLOW,
+			steps: [ 'plans', 'user', 'domains' ],
```

Comment:
Is there a rationale for having a different order from the rest of the flows? It might seem good to delay the `user` step, but last time we tested this (we tried putting the user step as the last step) it didn't actually improve performance.

The other thing is that `domains` is supposed to come before `plans` because otherwise we end up showing you plans that are ineligible for a custom domain. For example what happens if the user chooses the free plan and then they go ahead and pick a domain? Better to have the user pick the domain first and then we can only show plans which allow custom domains.

Reply:
>Is there a rationale for having a different order from the rest of the flows?

I feel a good idea to show the plans grid with the features before asking the user for credentials.

>The other thing is that domains is supposed to come before plans because otherwise we end up showing you plans that are ineligible for a custom domain. For example what happens if the user chooses the free plan and then they go ahead and pick a domain? Better to have the user pick the domain first and then we can only show plans which allow custom domains.

I see. Then maybe we can skip the domain selection altogether? I don't think it's a highlight for hosting services and we're creating a new flow anyway -- shouldn't be a hard sell. Plus, they can buy that later when creating a site for their business plan.

Reply:
> I feel a good idea to show the plans grid with the features before asking the user for credentials.

Fair enough. When this PR goes ready for review and Martech take a look they might have a better idea about whether there are gotchas with not having `user` at the front. But I don't see anything wrong with the idea in principle.

> Then maybe we can skip the domain selection altogether?

I think so. As a builder when I'm spinning up some resources on a new hosting provider I'm not expecting to have to think about domain name before I can use it.